src/ducks/sharing/index.js

```diff
@@ -0,0 +1,173 @@
+export const filterSharedDocuments = (ids, doctype) =>
+  findPermSets(ids, doctype).then(sets => sets.map(set => set.attributes.permissions.collection.values[0]))
+
+export const findPermSet = (id, doctype) =>
+  findPermSets([id], doctype).then(sets => sets.length === 0 ? undefined : sets[0])
+
+// TODO: move this to cozy-client-js
+// there is cozy.client.files.getCollectionShareLink already, but I think that
+// there is a need of a bit of exploratory work and design for that API...
+
+export const findPermSets = (ids, doctype) =>
+  cozy.client.fetchJSON('GET', `/permissions/doctype/${doctype}/sharedByLink`)
+    .then(sets => sets.filter(set => {
+      const perm = set.attributes.permissions.files
+      return perm.type === doctype && ids.find(id => perm.values.indexOf(id) !== -1) !== undefined
+    }))
+
+export const createPermSet = (id, doctype) =>
+  cozy.client.fetchJSON('POST', `/permissions?codes=email`, {
+    data: {
+      type: 'io.cozy.permissions',
+      attributes: {
+        permissions: {
+          files: {
+            type: 'io.cozy.files',
+            verbs: ['GET'],
+            values: [id]
+          }
+        }
+      }
+    }
+  })
+
+export const deletePermSet = (setId) =>
+  cozy.client.fetchJSON('DELETE', `/permissions/${setId}`)
+
+export const getShareLink = (id, perms) =>
+  `${window.location.origin}/public?sharecode=${perms.attributes.codes.email}&id=${id}`
+
+export class ShareModal extends React.Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      loading: true,
+      creating: false,
+      permissions: null,
+      active: false,
+      copied: false
+    }
+  }
+
+  componentDidMount () {
+    const { id } = this.props.document
+    findPermSet(id, 'io.cozy.files')
+      .then(permissions => {
+        if (permissions !== undefined) {
+          this.setState({ loading: false, permissions, active: true })
+        } else {
+          this.setState({ loading: false })
+        }
+      })
+      .catch((err) => this.onError(err))
+  }
+
+  toggleShareLink (active) {
+    const { id } = this.props.document
+    if (active) {
+      this.setState({ creating: true })
+      createPermSet(id, 'io.cozy.files')
+        .then(permissions => this.setState({ active, permissions, creating: false }))
+        .catch((err) => this.onError(err))
+    } else {
+      const setId = this.state.permissions._id
+      this.setState({ active, permissions: null })
+      deletePermSet(setId)
+        .catch((err) => this.onError(err))
+    }
+  }
+
+  onError (err) {
+    console.error(err)
+    this.props.onClose()
+  }
+
+  render () {
+    const { t } = this.context
+    const { onClose } = this.props
+    const { loading, active, creating, permissions } = this.state
+    return (
+      <Modal
+        title={t('share.title')}
+        secondaryAction={onClose}
+      >
+        <Tabs initialActiveTab='link'>
+          <TabList className={styles['pho-share-modal-tabs']}>
```

Comment:
Remove the prefix `pho-` here so it matches the style `share-modal-tabs` in your CSS and we're good. 😄